server/src/main/java/org/opensearch/index/snapshots/blobstore/RemoteStoreShardShallowCopySnapshot.java

```diff
@@ -0,0 +1,326 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.snapshots.blobstore;
+
+import org.opensearch.OpenSearchParseException;
+import org.opensearch.core.ParseField;
+import org.opensearch.core.xcontent.ToXContentFragment;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Remote Store based Shard snapshot metadata
+ *
+ * @opensearch.internal
+ */
+public class RemoteStoreShardShallowCopySnapshot implements ToXContentFragment {
+
+    private final String snapshot;
+    private final long indexVersion;
+    private final long startTime;
+    private final long time;
+    private final int totalFileCount;
+    private final long totalSize;
+    private final long primaryTerm;
+    private final long commitGeneration;
+    private final String remoteStoreRepository;
+    private final String repositoryBasePath;
+    private final String indexUUID;
+    private final List<String> fileNames;
```

Comment:
I don't see any `version` field which will help to evolve this file in future.

Reply:
We still can use the version object from input and output stream while reading the file, so should still be fine.